_drafts/gdbundle.md

```diff
@@ -0,0 +1,549 @@
+```
+
+Simple, right? I've opted to use `pyproject.toml` instead of `setup.py` because
+I wanted to embrace the future of standardized packaging, but note that these
+packages will work with Python 2 and 3, so it shouldn't matter.
+
+### How It Works
+
+Let's look at the code in `gdbundle_example/gdb_loader.py`
+
+```python
+import gdb
+import os
+
+PACKAGE_DIR = os.path.dirname(__file__)
+
+SCRIPT_PATHS = [
+    [PACKAGE_DIR, 'scripts', 'example_gdb.gdb'],
+    [PACKAGE_DIR, 'scripts', 'example_gdb.py']
+]
+
+def _abs_path(path):
+    return os.path.abspath(os.path.join(*path))
+
+def gdbundle_load():
+    for script_path in SCRIPT_PATHS:
+        gdb.execute("source {}".format(_abs_path(script_path)))
+```
+
+Notice the `gdbundle_load` function which `gdbundle` will call directly. When
+this is called, it takes each script file located in `scripts/` and tells GDB to
+`source` it. Notice that both GDB Python and GDB Command scripts work!
+
+By using `gdb.execute("source <script>")`, plugins can source GDB scripts that
+were never designed with gdbundle in mind or designed to be distributed.
+
+## A Few More Examples
+
+Given the flexibility of gdbundle itself, it can allow developer to package up
+any of the already existing GDB scripts and make installation easier for
+everyone.
+
+Here are a few more examples of plugins I've created that wrap popular packages
+that I use frequently.
+
+- [gdbundle-gdb-dashboard](https://github.com/memfault/gdbundle-gdb-dashboard) -
+  gdbundle plugin for
+  [cyrus-and/gdb-dashboard](https://github.com/cyrus-and/gdb-dashboard).
+- [gdbundle-PyCortexMDebug](https://github.com/memfault/gdbundle-PyCortexMDebug) -
+  gdbundle plugin for
+  [bnahill/PyCortexMDebug](https://github.com/bnahill/PyCortexMDebug).
+- [gdbundle-voltron](https://github.com/memfault/gdbundle-voltron) - gdbbundle
+  plugin for [snare/voltron](https://github.com/snare/voltron). This example is
+  my personal favorite because the install script is particularly sketchy but
+  you can simply run `pip install gdbundle-voltron` and you're up and running!
+
+## Summary of Benefits of using gdbundle
+
+There are a handful of indisputable benefits of using gdbundle.
+
+1. Just `pip install gdbundle-<plugin-name>`. No more manually editing your
+   `~/.gdbinit` in specific ways depending on the extension.
+2. It **enables** developers to use virtual environments (and encourages it!)
+   without the need for each plugin to
+   [mangle `sys.path`](https://github.com/snare/voltron/blob/master/voltron/entry.py#L26-L34)
+   in creative ways and install native packages using `apt` or `brew`.
+3. Personal projects and team projects can have project-specific
+   `requirements.txt` and `.gdbinit` files. With these two in place, a new
+   developer would just need to `pip install -r requirements.txt` and they'll
+   have everything they need to start using the plugins.
+4. Discoverability. Want to find out what new gdbundle packages exist? Just go
```

Comment:
Will you publish a few of your examples here before publishing?

Reply:
Yes, I'll publish them today and patch up the readme's and repo links.